### PR TITLE
Add tutorial for ROS 2 galactic [12701]

### PR DESCRIPTION
**Pull request by jparisu**
Signed-off-by: jparisu <javierparis@eprosima.com>

### CODE REVIEW

File: docs/rst/ros/galactic/galactic.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+.. figure:: /rst/figures/screenshots/usage_example/init_domain.png
+    :align: center
+
+Execute ROS 2 demo nodes with statistics
+----------------------------------------
+
+To execute ROS 2 nodes with statistics two configurations must be taken into account:
+
+- The middleware used must be Fast DDS, set by an environmental variable.
+- To activate the report of statistic data, Fast DDS requires to use an environmental variable setting up those
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:55]:
```suggestion
- In order to activate the publication of statistical data, Fast DDS requires an environment variable specifying those
```

File: docs/rst/ros/galactic/galactic.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    statistic kinds that are going to be reported.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:56]:
```suggestion
    kinds of statistical data to be reported.
```

File: docs/rst/ros/galactic/galactic.rst [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+To execute each of the nodes, run the following commands in different terminals:
+
+.. code-block:: bash
+
+    export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+    export FASTDDS_STATISTICS="HISTORY_LATENCY_TOPIC;SUBSCRIPTION_THROUGHPUT_TOPIC;PHYSICAL_DATA_TOPIC" # add as many statistics topic as wanted
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:63]:
```suggestion
    export FASTDDS_STATISTICS="HISTORY_LATENCY_TOPIC;SUBSCRIPTION_THROUGHPUT_TOPIC;PHYSICAL_DATA_TOPIC" # add as many statistics topics as wanted
```

File: docs/rst/ros/galactic/galactic.rst [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    ros2 run demo_nodes_cpp listener
+
+.. code-block:: bash
+
+    export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+    export FASTDDS_STATISTICS="HISTORY_LATENCY_TOPIC;SUBSCRIPTION_THROUGHPUT_TOPIC;PHYSICAL_DATA_TOPIC" # add as many statistics topic as wanted
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:69]:
```suggestion
    export FASTDDS_STATISTICS="HISTORY_LATENCY_TOPIC;SUBSCRIPTION_THROUGHPUT_TOPIC;PHYSICAL_DATA_TOPIC" # add as many statistics topics as wanted
```

File: docs/rst/ros/galactic/galactic.rst [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    ros2 run demo_nodes_cpp talker
+
+Remember to source your `ROS 2 installation
+<https://docs.ros.org/en/galactic/Installation/Ubuntu-Development-Setup.html?highlight=source#source-the-setup-script>`_
+before every ros2 command.
+
+Monitoring network
+------------------
+
+Now you must see in the Fast DDS Monitor two new Participants.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:79]:
```suggestion
Now you should see in the DDS panel two new Participants.
```

File: docs/rst/ros/galactic/galactic.rst [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+.. figure:: /rst/figures/screenshots/galactic_tutorial/participants.png
+    :align: center
+
+Alias
+^^^^^
+
+Participants in ROS 2 are called :code:`\ `.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:87]:
```suggestion
Participants in ROS 2 are named :code:`\ ` by default.
```

File: docs/rst/ros/galactic/galactic.rst [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+In order to differentiate them you could :ref:`change_alias`.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:88]:
```suggestion
In order to differentiate them one could :ref:`change_alias`.
```

File: docs/rst/ros/galactic/galactic.rst [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+The :code:`talker` would be the one with one writer, and the :code:`listener` with a reader.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:89]:
```suggestion
The :code:`talker` would be the one with one writer, and the :code:`listener` the one with a reader.
```

File: docs/rst/ros/galactic/galactic.rst [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+
+.. figure:: /rst/figures/screenshots/galactic_tutorial/alias.png
+    :align: center
+
+Physical data
+^^^^^^^^^^^^^
+
+In order to see the information about the host and the physical context where every node is running, go to
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:97]:
```suggestion
In order to see the information of the host and the physical context where every node is running, go to
```

File: docs/rst/ros/galactic/galactic.rst [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+the :ref:`left_panel` and activate the :ref:`physical_panel`.
+There are shown the host, user and process of each node.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:99]:
```suggestion
There, the host, user and process of each node are displayed.
```

File: docs/rst/ros/galactic/galactic.rst [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+
+.. figure:: /rst/figures/screenshots/galactic_tutorial/physical.png
+    :align: center
+
+Statistical data
+^^^^^^^^^^^^^^^^
+
+To show statistical data about the communication between the :code:`talker` and the :code:`listener`.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:107]:
```suggestion
To show statistical data about the communication between the :code:`talker` and the :code:`listener`,
```

File: docs/rst/ros/galactic/galactic.rst [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+Follow the steps to :ref:`tutorial_create_dynamic_series` and see this statistical data graphically in a real
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:108]:
```suggestion
follow the steps to :ref:`tutorial_create_dynamic_series` and plot this statistical data in a real
```

File: docs/rst/ros/galactic/galactic.rst [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+time chart.
+
+.. figure:: /rst/figures/screenshots/galactic_tutorial/statistics.png
+    :align: center
+
+Introspect metatraffic topics
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Fast DDS Monitor filters by default the metatraffic topics and the endpoints related to them so the user
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:117]:
```suggestion
Fast DDS Monitor filters by default the topics used for sharing metatraffic and the endpoints related to them so the user
```

File: docs/rst/ros/galactic/galactic.rst [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+can inspection its network easily.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:118]:
```suggestion
can inspect its network easily.
```

File: docs/rst/ros/galactic/galactic.rst [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+These topics are the ones that ROS 2 uses for discovery and configuration, such as `ros_discovery_info`.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:119]:
```suggestion
These topics are the ones that ROS 2 uses for discovery and configuration purposes, such as `ros_discovery_info`,
```

File: docs/rst/ros/galactic/galactic.rst [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+And those used by Fast DDS to report statistical data.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:120]:
```suggestion
 as well as those used by Fast DDS to report statistical data.
```

File: docs/rst/ros/galactic/galactic.rst [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+
+.. todo:
+
+    Add link to metatraffic section and metatraffic show button
+
+In order to see these topics in the monitor, click .
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:126]: click reference is missing

File: docs/rst/ros/galactic/galactic.rst [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+Now, these topics will be shown in the logical panel, and the Readers and Writers associated to them under their
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:127]:
```suggestion
Now, these topics are shown in the logical panel, and also the Readers and Writers associated to them under their
```

File: docs/rst/ros/galactic/galactic.rst [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+respective Participants.
+
+.. figure:: /rst/figures/screenshots/galactic_tutorial/metatraffic.png
+    :align: center
+
+Video Tutorial
+==============
+
+There is a `video tutorial <https://www.youtube.com/watch?v=OYibnUnMIlc>`_ briefly showing the tutorial
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:136]:
```suggestion
There is a `video tutorial <https://www.youtube.com/watch?v=OYibnUnMIlc>`_ going through the steps
```

Comment (juanlofer-eprosima) [reply docs/rst/ros/galactic/galactic.rst:136]: Video link to be modified after reuploading new version

File: docs/rst/ros/galactic/galactic.rst [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
+that has been explained in this section.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/galactic/galactic.rst:137]:
```suggestion
described in this section.
```

File: docs/rst/ros/ros.rst [hunk 21]
```diff
@@ -0,0 +1,23 @@
+.. _ros_section:
+
+###########################
+Fast DDS Monitor with ROS 2
+###########################
+
+Fast DDS Monitor is a very useful tool to monitor and study a ROS 2 network.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/ros.rst:7]:
```suggestion
Fast DDS Monitor is a useful tool for monitoring and studying a ROS 2 network.
```

File: docs/rst/ros/ros.rst [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+The automatic discovery of entities in a local network allows to easily see the different Participants that
+are running, as its endpoints, the topics that each one is using, and even the network interfaces that are being
+used to communicate each other.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/ros.rst:10]:
```suggestion
to communicate with each another.
```

File: docs/rst/ros/ros.rst [hunk 23]
```diff
@@ -0,0 +1,23 @@
+.. _ros_section:
+
+###########################
+Fast DDS Monitor with ROS 2
+###########################
+
+Fast DDS Monitor is a very useful tool to monitor and study a ROS 2 network.
+The automatic discovery of entities in a local network allows to easily see the different Participants that
+are running, as its endpoints, the topics that each one is using, and even the network interfaces that are being
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/ros.rst:9]:
```suggestion
are running, as its endpoints, the topics that each one is using, and even the network interfaces they are using
```

File: docs/rst/ros/ros.rst [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+used to communicate each other.
+Additionally, you could receive statistics data from every endpoint in the network.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/ros.rst:11]:
```suggestion
Additionally, one could receive statistical data from every endpoint in the network.
```

File: docs/rst/ros/ros.rst [hunk 25]
```diff
@@ -0,0 +1,23 @@
...
+This data is very useful to analyze the performance and seek any possible communication problem in the network.
+
+ROS 2 relies on DDS communication protocol to communicate different nodes.
+There are different RMWs (ROS MiddleWare) available in the latest ROS 2 versions.
+Just be sure to use Fast DDS as your ROS 2 middleware in order to receive statistic data and monitor your network.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/ros.rst:16]:
```suggestion
Make sure you are using Fast DDS as your ROS 2 middleware in order to receive statistical data and monitor your network.
```

File: docs/rst/ros/ros.rst [hunk 26]
```diff
@@ -0,0 +1,23 @@
...
+
+In the following sections are described how to use Fast DDS with Statistics active in different ROS2 versions.
```
Comment (juanlofer-eprosima) [inline docs/rst/ros/ros.rst:18]:
```suggestion
In the following sections, instructions for using Fast DDS with Statistics in different ROS2 versions are provided, as well as a brief tutorial demonstrating the integration of Fast DDS Monitor with ROS2 in a simple scenario.
```

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

File: docs/rst/ros/galactic/galactic.rst [hunk 27]
```diff
@@ -0,0 +1,138 @@
...
+
+Execution
+=========
+
+We are going to recreate a simple DDS network with one :code:`talker` and one :code:`listener` from ROS 2 demo nodes.
+
+Execute Fast DDS Monitor
+------------------------
+
+Initiate Fast DDS Monitor by the executable created by the installation process.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:43]: :+1:

File: docs/rst/ros/galactic/galactic.rst [hunk 28]
```diff
@@ -0,0 +1,138 @@
...
+Once into the Fast DDS Monitor window, start a monitor in domain :code:`0` (default domain).
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:44]: :+1:

File: docs/rst/ros/galactic/galactic.rst (hunk 1 above)
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:55]: :+1:

File: docs/rst/ros/galactic/galactic.rst (hunk 2 above)
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:56]: :+1:

File: docs/rst/ros/galactic/galactic.rst (hunk 4 above)
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:69]: :+1:

File: docs/rst/ros/galactic/galactic.rst (hunk 7 above)
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:88]: :+1:

File: docs/rst/ros/galactic/galactic.rst (hunk 9 above)
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:97]: :+1:

File: docs/rst/ros/galactic/galactic.rst (hunk 11 above)
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:107]: :+1:

File: docs/rst/ros/galactic/galactic.rst (hunk 12 above)
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:108]: :+1:

File: docs/rst/ros/galactic/galactic.rst (hunk 13 above)
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:117]: :+1:

File: docs/rst/ros/galactic/galactic.rst (hunk 15 above)
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:119]: :+1:

File: docs/rst/ros/galactic/galactic.rst (hunk 18 above)
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:127]: :+1:

File: docs/rst/ros/galactic/galactic.rst (hunk 20 above)
Comment (jparisu) [reply docs/rst/ros/galactic/galactic.rst:137]: :+1:

File: docs/rst/ros/ros.rst [hunk 29]
```diff
@@ -0,0 +1,23 @@
+.. _ros_section:
+
+###########################
+Fast DDS Monitor with ROS 2
+###########################
+
+Fast DDS Monitor is a very useful tool to monitor and study a ROS 2 network.
+The automatic discovery of entities in a local network allows to easily see the different Participants that
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/ros/ros.rst:8]: Use capital letter for Participants to keep the documentation format

File: docs/rst/ros/ros.rst (hunk 23 above)
Comment (jparisu) [reply docs/rst/ros/ros.rst:9]: :+1:

File: docs/rst/ros/ros.rst (hunk 22 above)
Comment (juanlofer-eprosima) [reply docs/rst/ros/ros.rst]: sorry, I meant "with each other"

File: docs/rst/ros/ros.rst (hunk 24 above)
Comment (juanlofer-eprosima) [reply docs/rst/ros/ros.rst]: impersonal?

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]
Please check the open threads

**Comment by juanlofer-eprosima** [comment]
LGTM

**Pull request merged**